browser/ui/views/side_panel/ai_chat_side_panel_web_view.h

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#ifndef BRAVE_BROWSER_UI_VIEWS_SIDE_PANEL_AI_CHAT_SIDE_PANEL_WEB_VIEW_H_
+#define BRAVE_BROWSER_UI_VIEWS_SIDE_PANEL_AI_CHAT_SIDE_PANEL_WEB_VIEW_H_
+
+#include <memory>
+
+#include "base/memory/weak_ptr.h"
+#include "chrome/browser/ui/views/side_panel/side_panel_web_ui_view.h"
+
+class AIChatUI;
+class Profile;
+
+// A custom web view to set focus correctly when the side panel is shown.
+class AIChatSidePanelWebView : public SidePanelWebUIViewT<AIChatUI> {
+ public:
+  // Factory method to create and configure an AIChatSidePanelWebView.
+  // If `is_tab_associated` is true, the side panel will be related
+  // to the active tab and will change conversation when the tab navigates.
+  static std::unique_ptr<views::View> CreateView(base::WeakPtr<Profile> profile,
+                                                 bool is_tab_associated,
+                                                 SidePanelEntryScope& scope);
+
+  AIChatSidePanelWebView(
+      SidePanelEntryScope& scope,
+      std::unique_ptr<WebUIContentsWrapperT<AIChatUI>> contents_wrapper);
+  ~AIChatSidePanelWebView() override;
+
+  // Disable copy and assign.
+  AIChatSidePanelWebView(const AIChatSidePanelWebView&) = delete;
+  AIChatSidePanelWebView& operator=(const AIChatSidePanelWebView&) = delete;
+
+ private:
+  // This callback is invoked multiple times, so we need to ensure that
+  // focus is set only once with `should_focus_`.
+  void OnShow();
+
+  bool should_focus_ = true;
```

Comment:
nit: a little comment about this field.

Reply:
This class was moved and it seems like this variable is mentioned in the above comment for `OnShow` but I've added an attached comment here